### DISCUSSION TITLE
Can't interact with page rendered by external rewrite

### DISCUSSION THREAD

**Discussion by gabrielEloy** [vercel/next.js#50053] [Help]
### Summary

I have an application in which I need to render an app hosted in another URL. Both apps are owned by me. Although I'm able to correctly render the external website using rewrites, I'm not able to interact with it. Whenever I click any button on my website, nothing happens.

the App I'm trying to create an external rewrite for is a create next app. The only file I changed on it is `src/pages/index` and it looks like this:

```ts

import { useState } from 'react'



export default function Home() {
  const [localValue, setLocalValue] = useState<any>(null)
  const [logs, setLogs] = useState<any[]>([]);
  
  const handleRetrieveValue = () => {
    const retrievedValue = localStorage.getItem('kronos');
    setLocalValue(retrievedValue )
    addToRetrievalsLog({date: new Date(), value: retrievedValue})
  }

  const addToRetrievalsLog = (log: any) => {
    setLogs([log, ...logs]);
  }
  
  return (
    <>
    <div>
      <h1>retrieve values test</h1>
      <button onClick={handleRetrieveValue}>retrieve</button>
      <p>value: </p>
      <p>{String(localValue)}</p>
      <h3>logs</h3>
      <ul>
        {logs.map(log => <li key={log.date.getTime()}>
          <span>value: {String(log.value)} - {log.date.toString()}</span>
        </li>)}
      </ul>
    </div>
    </>
  )
}

```

I have a video of this application working on its own URL:

https://github.com/vercel/next.js/assets/37480915/dc9a7035-5ed6-4d5f-a8ce-6134f605ed8e

and Here's a video of it not working inside my rewritten URL:

https://github.com/vercel/next.js/assets/37480915/c9c3818e-5de0-4a8b-9fe6-36a016429673

I've removed all CSP on the app in both apps, so that shouldn't be what's causing the issue

**Comment by Toriong** [comment]
Dealing with the same issue. What does your `next.config.file` look like? Here's mine: 

```module.exports = {
  reactStrictMode: true,
  images: {
    domains: [
      'catalog.galacticpolymath.com',
      'gp-catalog.vercel.app',
      'i3.ytimg.com',
      'storage.googleapis.com',
      'into-the-dark.vercel.app',
      'echo-galactic-polymath.vercel.app'
    ],
  },
  async rewrites() {
    return [
      {
        source: '/apps/echoSim',
        destination: 'https://echo-jade.vercel.app'
      },
    ]
  }
};
```

**Reply by Toriong** [reply]
This discussion might help. It looks like this might be the intended behavior of rewriting to an external site: https://github.com/vercel/next.js/discussions/20650
